public/js/types.js

```diff
@@ -13,8 +13,8 @@
  */
 export type Source = {
   id: string,
-  url?: string,
-  sourceMapURL?: string
+  url: string,
+  sourceMapURL: string
```

Comment:
Why these changes? This is wrong. A source might not have a URL, and definitely will not have a sourceMapURL most of the time.